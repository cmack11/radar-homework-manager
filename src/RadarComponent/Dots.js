import React, { Component } from 'react';
import moment from 'moment'
import util from './utils.js'
import PropTypes from 'prop-types';
import Dot from './Dot.js';
import { connect } from 'react-redux';


const mapDispatchToProps = dispatch => ({
})

const mapStateToProps = state => {
  return {
    id: state.user.user_id,
    types : state.assignment.typesDict,
  }
}

class Dots extends Component {


	constructor(props) {
		super();
		let state = {
			width:window.innerWidth,
			height:window.innerHeight,
		}
		this.state = state;
		this.variableRowSizeFactor = .8;


	}

	componentWillReceiveProps(nextProps) {
		if(nextProps.view.dotsView.width !== this.state.width)
			this.setState({width:nextProps.view.dotsView.width})

		if(nextProps.view.dotsView.height !== this.state.height)
			this.setState({height:nextProps.view.dotsView.height})
	}

	componentDidMount() {
		//window.addEventListener('resize', this.updateDimensions.bind(this));
		/*setInterval(() => {
			for(let i = 0; this.dotsObjs && i < this.dotsObjs.length; i++)
				for(let j = 0; j < this.dotsObjs[i].length; j++)

		},4000)	*/
	}



	//Make Dot Componenent so it can do its own opacity management
	makeDot(dot) {
		if(!this.view.dots || !this.view.dots.center) return null;
		var distanceFromCenter = dot.distanceFromCenter;
		var angle = dot.startAngle + dot.angle;
		var point = util.polarToCartesian(this.view.dots.center.x,this.view.dots.center.y,distanceFromCenter,angle);
		if(!dot.r) dot.r = this.view.dots.radius;

		return <Dot id={dot.assignment.name+' '+dot.assignment.type+'outer'} center={point} radius={dot.r} fill={dot.color}
		onMouseDown={this.onMouseDownDot.bind(this)}
		dot={dot}
		intersectsLine={this.props.intersectsLine}
		animateFades={true}
		clickable={!this.props.view.disable}
		/>
	}

	fillDotsObjs(subjects) {
		if(!subjects) return;

		this.dotsObjs = [];
		let i, j, angle;
		if(subjects && subjects.length)
			angle = 360/subjects.length;
		else
			angle = 0;

		for(i = 0; subjects && i < subjects.length; i++) {
			this.dotsObjs[i] = [];
			for(j = 0; j < subjects[i].assignments.length; j++) {
				let assignment = subjects[i].assignments[j];
				let distanceFromCenter = this.props.getDistanceFromCenter(assignment);
				if(distanceFromCenter < -2) continue;
				if(distanceFromCenter <= -1) continue;

				let color = 'white';
				if(this.props.types && this.props.types[assignment.type_id])
					color = this.props.types[assignment.type_id].color;
				this.dotsObjs[i].push({
					key:(subjects[i].name+'/'+assignment.name+'/'+moment(assignment.dueDate).format('YYYY-MM-DD HH:mm')),
					distanceFromCenter:distanceFromCenter,
					radius:15,
					startAngle:i*angle,
					maxAngle:angle,
					angle:.5*angle,
					color:color,
					assignment:assignment
				})
			}
		}
	}

	getDotRows(dots, fixed, numSteps) {
		if(!dots || dots.length === 0 || !this.view.radar) return null;
		dots.sort(this.compareDots);

		if(fixed && !numSteps)
			numSteps = 10;

		var outerBoundary, innerBoundary, currRow = 0, rows, step, start;

		if(fixed) {
			start = 0;
			step = this.view.radar.radius/numSteps;
			outerBoundary = this.view.radar.radius;
			innerBoundary = outerBoundary-step;
			rows = [[]];
		} else {
			start = 1;
			step = this.view.dots.radius*2;
			outerBoundary = dots[0].distanceFromCenter;
			innerBoundary = outerBoundary - step;
			rows = [[dots[0]]];
			currRow = 0;
		}

		for(let i = start; i < dots.length; i++) {
			//If in boundry, add it to the row
			if(innerBoundary < dots[i].distanceFromCenter && dots[i].distanceFromCenter <= outerBoundary) {
				//Add the dot to the current row
				rows[currRow].push(dots[i]);
				if(!fixed) {
					innerBoundary = dots[i].distanceFromCenter - step;
					//Decrement the next step by a factor of .9 to curb number of rows
					step *= this.variableRowSizeFactor;
				}
				if(outerBoundary < 0) break;
			} else {
				if(fixed){//Increase the boundry a fixed amount and create new row
					i--;
					outerBoundary = innerBoundary;
					innerBoundary -= step;
					currRow++;
					rows[currRow] = [];
				} else {//Set new boundry at next dot, create new row
					outerBoundary = dots[i].distanceFromCenter
					innerBoundary = outerBoundary - this.view.dots.radius*2;
					if(outerBoundary < 0) break;
					currRow++;
					rows[currRow] = [dots[i]];
					step = this.view.dots.radius*2;
				}
			}
		}

		return rows;
	}

	disperseRow(row, maxAngle) {
		if(!row || !row.length) return;

		let rowAngles = [], step;
		if(this.props.subjects.length > 1)
			step = maxAngle / (row.length+1);
		else
			step = maxAngle / (row.length);

		//Fill an array of the angles that will be "claimed" by the dots in the row
		for(var k = 0; k < row.length; k++)
			rowAngles.push((k+1)*step);
		//if odd amount of dots, do first and then only worry about even
		var rowAngleIndex, rowIndex, close = (row.length % 2 !== 0);
		while(row.length > 0) {
			if(row.length % 2 === 0) close = !close;//Switch doing close every two times

			rowAngleIndex = Math.floor(rowAngles.length/2);
			if(close)//Take dot from the back (dots close to center)
				rowIndex = row.length-1;
			else//Take dot from the front (dots close to the edge)
				rowIndex = 0;

			//Set the dot's angle and radius and then remove it and the angle it claimed from the list
			row[rowIndex].angle = rowAngles[rowAngleIndex];
			row.splice(rowIndex,1);
			rowAngles.splice(rowAngleIndex,1);
		}
	}

	compareDots(a,b) {
		if(a.distanceFromCenter < b.distanceFromCenter)
			return 1;
		if(a.distanceFromCenter > b.distanceFromCenter)
			return -1;
		return 0;
	}

	shuffleDots(dots) {
		if(dots.length < 1) return;

		let rows = this.getDotRows(dots);

		for(let i = 0; rows && i < rows.length; i++) {
			let row = rows[i];
			if(row.length === 0) continue;

			this.disperseRow(row,row[0].maxAngle);

		}
	}


	onClickDot(e) {
		let dotsGroup = document.getElementById('dotsGroup');
		dotsGroup.insertBefore(e.target,null);
		//move dot viewer to radar so it can sit on top
		//let dotViewer = document.getElementById('dotViewer');
		//dotsGroup.insertBefore(dotViewer,null);
		//let state = this.state;
		this.props.setClickedDot({
			x:e.target.getAttribute('cx'),
			y:e.target.getAttribute('cy'),
			fill:e.target.getAttribute('fill'),
			id:e.target.id,
			assignment:this.draggedDot.assignment
		})
		//this.setState(state);
	}

	pauseEvent(e){
	    if(e.stopPropagation) e.stopPropagation();
	    if(e.preventDefault) e.preventDefault();
	    e.cancelBubble=true;
	    e.returnValue=false;
	    return false;
	}

	onMouseDownDot(e,dotObj) {
		this.pauseEvent(e);
		if(this.props.view.disable) return;
		this.draggedDot = {
			dot:e.target,
			assignment:dotObj.assignment,
			moved:false,
			mousemove:this.onMouseMoveDot.bind(this),
			mouseup:this.onMouseUpDot.bind(this),
			onclick:this.onClickDot.bind(this),
			original:{cx:e.target.getAttribute('cx'), cy:e.target.getAttribute('cy')}
		};
		this.props.setDraggedDot(this.draggedDot)
		window.addEventListener('mouseup',this.draggedDot.mouseup);
		window.addEventListener('mousemove',this.draggedDot.mousemove);


	}

	onMouseMoveDot(e) {
		this.pauseEvent(e);
		this.draggedDot.moved = true;
		this.draggedDot.dot.setAttribute('visibility','hidden');
		window.removeEventListener('mousemove',this.draggedDot.mousemove)
	}

	onMouseUpDot(e) {
		this.props.setDraggedDot(null)

		window.removeEventListener('mouseup',this.draggedDot.mouseup);
		//Doesn't work anymore, needs to be converted to work in DraggedDot component

		if(this.draggedDot.moved) {
			this.draggedDot.dot.setAttribute('visibility','visible');
		} else {
			window.removeEventListener('mousemove',this.draggedDot.mousemove)
			this.onClickDot(e);
		}
		this.draggedDot = null;
	}


	render() {
		this.view = this.props.dims;
		var subjects = this.props.subjects;


		this.fillDotsObjs(subjects);

		this.dotsElems = [];
		for(let i = 0; this.dotsObjs && i < this.dotsObjs.length; i++) {
			this.shuffleDots(this.dotsObjs[i]);
			for(let j = 0; j < this.dotsObjs[i].length; j++)
				this.dotsElems.push(this.makeDot(this.dotsObjs[i][j]));
		}

		return (
			<svg id="dotsGroup" width={this.state.width} height={this.state.height} stroke="black" strokeWidth="2">
				{this.dotsElems}
			</svg>

		)
	}
}

Dots.propTypes = {
  	view: PropTypes.object.isRequired,
  	subjects: PropTypes.array.isRequired,
  	getDistanceFromCenter: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(Dots);

//need to find a way to have the dragged dot be on top of the spinline, but stationary dots be below
//And also have dotviewer be on top of spinline
