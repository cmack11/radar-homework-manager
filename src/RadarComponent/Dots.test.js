import React from 'react';
import {Dots} from './Dots.js';
import { shallow, mount, render } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {subjects1} from '../fakeData.js';
import moment from 'moment'

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() })
  jest.useFakeTimers();
});

const subjects = [
  {
    name:'Subject #1',
    color:'red',
    assignments:[
      {
        name:"Assignment #1",
        type:"Problem Set",
        dueDate:moment('2018-09-12 12:00:00.000'),
      subject:"Subject #1",
      },{
        name:"Assignment #2",
        type:"Assignment",
        dueDate:moment('2018-09-13 14:00:00.000'),
        subject:"Subject #1",
      }
    ]
  },
  {
    name:'Subject #2',
    color:'blue',
    assignments:[]
  }
]

const getDistanceFromCenter = function() {return 1};
const view = {}
const dims = {};

const dotObj = {

}

/*<Dots  subjects={this.state.subjects} 
          getDistanceFromCenter={this.getDistanceFromCenter.bind(this)} 
          view={this.state.view} dims={this.view} //bad
          intersectFunctions={intersectFuncs}
          intersectsLine={this.intersectsLine.bind(this)}
          setClickedDot={this.setClickedDot.bind(this)}
          setDraggedDot={this.setDraggedDot.bind(this)}/>,*/

describe('the initialization of the component', () => {

  it('creates layer of dots', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('creates an empty dot layer', () => {
    const wrapper = shallow(
      <Dots  subjects={[]} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />
    );

    expect(wrapper).toMatchSnapshot();
  });

})

describe('componentWillReceiveProps', () => {

  it('should not change the layers dimensions', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.state.width = 100;
      dotsLayer.state.height = 200;
      let newProps = {
        view:{dotsView:{width:100,height:200}}
      };
      let stateSpy = jest.spyOn(dotsLayer,'setState')

      dotsLayer.componentWillReceiveProps(newProps);

      expect(dotsLayer.state.width).toEqual(100)
      expect(dotsLayer.state.height).toEqual(200)
      expect(stateSpy).not.toBeCalled();

  });

  it('should not change the layers dimensions', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.state.width = 400;
      dotsLayer.state.height = 500;
      let newProps = {
        view:{dotsView:{width:100,height:200}}
      };
      let stateSpy = jest.spyOn(dotsLayer,'setState')

      dotsLayer.componentWillReceiveProps(newProps);

      expect(dotsLayer.state.width).toEqual(100)
      expect(dotsLayer.state.height).toEqual(200)
      expect(stateSpy).toBeCalledTimes(2);

  });

})

describe('makeDot', () => {

  it('should return null because of missing dimensions', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {};
      let dot = {};

      let returnValue = dotsLayer.makeDot(dot);

      expect(returnValue).toMatchSnapshot();

  });

  it('should return null because of missing dimensions', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {dots:{}};
      let dot = {};

      let returnValue = dotsLayer.makeDot(dot);

      expect(returnValue).toMatchSnapshot();

  });

  it('should return a dot component', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {dots:{center:{x:50,y:50}}};
      let dot = {
        distanceFromCenter:10,
        startAngle:10,
        angle:30,
        assignment:{name:'Test Name',type:'Type'}
      };

      let returnValue = dotsLayer.makeDot(dot);

      expect(returnValue).toMatchSnapshot();

  });

  it('should return a dot component', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {dots:{center:{x:50,y:50}}};
      let dot = {
        distanceFromCenter:10,
        startAngle:10,
        angle:30,
        r:15,
        assignment:{name:'Test Name',type:'Type'}
      };

      let returnValue = dotsLayer.makeDot(dot);

      expect(returnValue).toMatchSnapshot();
  });

})

describe('fillDotsObjs', () => {

  it('should do nothing', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();

      let returnValue = dotsLayer.fillDotsObjs();

      expect(returnValue).not.toBeDefined();
  });

  it('should add no objects because there are no assignments', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      let testSubjects = [];

      dotsLayer.fillDotsObjs(testSubjects);

      expect(dotsLayer.dotsObjs).toHaveLength(0);
  });

  it('should convert the subjects assignments to dot objects', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      let testSubjects = subjects;

      dotsLayer.fillDotsObjs(testSubjects);

      expect(dotsLayer.dotsObjs).toHaveLength(subjects.length);
      for(let i = 0; i < subjects.length; i++) {
        expect(dotsLayer.dotsObjs[i]).toHaveLength(subjects[i].assignments.length);
      }
  });

  it('should add no objects because they are out of the time range', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={()=>{return -1}} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();

      expect(dotsLayer.dotsObjs).toHaveLength(subjects.length);
      for(let i = 0; i < subjects.length; i++) {
        expect(dotsLayer.dotsObjs[i]).toHaveLength(0);
      }
  });

  it('should add no objects because they are out of the time range', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={()=>{return -3}} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();

      expect(dotsLayer.dotsObjs).toHaveLength(subjects.length);
      for(let i = 0; i < subjects.length; i++) {
        expect(dotsLayer.dotsObjs[i]).toHaveLength(0);
      }
  });

  it('should convert the subjects assignments to well-formed dot objects', () => {
    let fullView = view;
    fullView.colors = {typeColors:{Assignment:'blue'}}
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();

      expect(dotsLayer.dotsObjs).toMatchSnapshot();
  });

})

describe('getDotRows', () => {

  it('should return null', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      let dots = null;

      let returnValue = dotsLayer.getDotRows(dots)

      expect(returnValue).toMatchSnapshot();
  });

  it('should return null', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      let dots = [];

      let returnValue = dotsLayer.getDotRows(dots)

      expect(returnValue).toMatchSnapshot();
  });

  it('should return null', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view.radar = null;
      let dots = [];

      let returnValue = dotsLayer.getDotRows(dots)

      expect(returnValue).toMatchSnapshot();
  });

  it('should make variable sized rows', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {
        dots:{radius:5},
        radar:{},
      }
      let dots = [
      {distanceFromCenter:5},
      {distanceFromCenter:10},
      {distanceFromCenter:15},
      {distanceFromCenter:25}
      ];

      let returnValue = dotsLayer.getDotRows(dots)

      expect(returnValue).toMatchSnapshot();
  });

  it('should make fixed sized rows', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {
        dots:{},
        radar:{radius:100},
      }
      let dots = [
      {distanceFromCenter:5},
      {distanceFromCenter:7},
      {distanceFromCenter:10},
      {distanceFromCenter:15},
      {distanceFromCenter:25},
      {distanceFromCenter:23}
      ];

      let returnValue = dotsLayer.getDotRows(dots,true,10)

      expect(returnValue).toMatchSnapshot();
  });

  it('should make fixed sized rows using the default amount of rows', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();
      dotsLayer.view = {
        dots:{},
        radar:{radius:100},
      }
      let dots = [
      {distanceFromCenter:5},
      {distanceFromCenter:7},
      {distanceFromCenter:10},
      {distanceFromCenter:15},
      {distanceFromCenter:25},
      {distanceFromCenter:23}
      ];

      let returnValue = dotsLayer.getDotRows(dots,true)

      expect(returnValue).toMatchSnapshot();
  });

})


/*describe(' ', () => {

  it('should ', () => {
    const wrapper = shallow(
      <Dots  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let dotsLayer = wrapper.instance();


  });

})*/



