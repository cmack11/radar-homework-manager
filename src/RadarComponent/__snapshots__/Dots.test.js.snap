// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fillDotsObjs should convert the subjects assignments to well-formed dot objects 1`] = `
Array [
  Array [
    Object {
      "angle": 90,
      "assignment": Object {
        "dueDate": "2018-09-12T17:00:00.000Z",
        "name": "Assignment #1",
        "subject": "Subject #1",
        "type": "Problem Set",
      },
      "color": "white",
      "distanceFromCenter": 1,
      "key": "Subject #1/Assignment #1/2018-09-12 12:00",
      "maxAngle": 180,
      "radius": 15,
      "startAngle": 0,
    },
    Object {
      "angle": 90,
      "assignment": Object {
        "dueDate": "2018-09-13T19:00:00.000Z",
        "name": "Assignment #2",
        "subject": "Subject #1",
        "type": "Assignment",
      },
      "color": "white",
      "distanceFromCenter": 1,
      "key": "Subject #1/Assignment #2/2018-09-13 14:00",
      "maxAngle": 180,
      "radius": 15,
      "startAngle": 0,
    },
  ],
  Array [],
]
`;

exports[`getDotRows should make fixed sized rows 1`] = `
Array [
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [
    Object {
      "distanceFromCenter": 25,
    },
    Object {
      "distanceFromCenter": 23,
    },
  ],
  Array [
    Object {
      "distanceFromCenter": 15,
    },
  ],
  Array [
    Object {
      "distanceFromCenter": 10,
    },
    Object {
      "distanceFromCenter": 7,
    },
    Object {
      "distanceFromCenter": 5,
    },
  ],
]
`;

exports[`getDotRows should make fixed sized rows using the default amount of rows 1`] = `
Array [
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [],
  Array [
    Object {
      "distanceFromCenter": 25,
    },
    Object {
      "distanceFromCenter": 23,
    },
  ],
  Array [
    Object {
      "distanceFromCenter": 15,
    },
  ],
  Array [
    Object {
      "distanceFromCenter": 10,
    },
    Object {
      "distanceFromCenter": 7,
    },
    Object {
      "distanceFromCenter": 5,
    },
  ],
]
`;

exports[`getDotRows should make variable sized rows 1`] = `
Array [
  Array [
    Object {
      "distanceFromCenter": 25,
    },
  ],
  Array [
    Object {
      "distanceFromCenter": 15,
    },
    Object {
      "distanceFromCenter": 10,
    },
    Object {
      "distanceFromCenter": 5,
    },
  ],
]
`;

exports[`getDotRows should return null 1`] = `null`;

exports[`getDotRows should return null 2`] = `null`;

exports[`getDotRows should return null 3`] = `null`;

exports[`makeDot should return a dot component 1`] = `
<Dot
  animateFades={true}
  center={
    Object {
      "x": 56.42787609686539,
      "y": 42.33955556881022,
    }
  }
  clickable={true}
  dot={
    Object {
      "angle": 30,
      "assignment": Object {
        "name": "Test Name",
        "type": "Type",
      },
      "distanceFromCenter": 10,
      "r": undefined,
      "startAngle": 10,
    }
  }
  id="Test Name Typeouter"
  onMouseDown={[Function]}
/>
`;

exports[`makeDot should return a dot component 2`] = `
<Dot
  animateFades={true}
  center={
    Object {
      "x": 56.42787609686539,
      "y": 42.33955556881022,
    }
  }
  clickable={true}
  dot={
    Object {
      "angle": 30,
      "assignment": Object {
        "name": "Test Name",
        "type": "Type",
      },
      "distanceFromCenter": 10,
      "r": 15,
      "startAngle": 10,
    }
  }
  id="Test Name Typeouter"
  onMouseDown={[Function]}
  radius={15}
/>
`;

exports[`makeDot should return null because of missing dimensions 1`] = `null`;

exports[`makeDot should return null because of missing dimensions 2`] = `null`;

exports[`the initialization of the component creates an empty dot layer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dots
    dims={Object {}}
    getDistanceFromCenter={[Function]}
    subjects={Array []}
    view={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "height": 768,
      "id": "dotsGroup",
      "stroke": "black",
      "strokeWidth": "2",
      "width": 1024,
    },
    "ref": null,
    "rendered": Array [],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "height": 768,
        "id": "dotsGroup",
        "stroke": "black",
        "strokeWidth": "2",
        "width": 1024,
      },
      "ref": null,
      "rendered": Array [],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`the initialization of the component creates layer of dots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dots
    dims={Object {}}
    getDistanceFromCenter={[Function]}
    subjects={
      Array [
        Object {
          "assignments": Array [
            Object {
              "dueDate": "2018-09-12T17:00:00.000Z",
              "name": "Assignment #1",
              "subject": "Subject #1",
              "type": "Problem Set",
            },
            Object {
              "dueDate": "2018-09-13T19:00:00.000Z",
              "name": "Assignment #2",
              "subject": "Subject #1",
              "type": "Assignment",
            },
          ],
          "color": "red",
          "name": "Subject #1",
        },
        Object {
          "assignments": Array [],
          "color": "blue",
          "name": "Subject #2",
        },
      ]
    }
    view={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "height": 768,
      "id": "dotsGroup",
      "stroke": "black",
      "strokeWidth": "2",
      "width": 1024,
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "height": 768,
        "id": "dotsGroup",
        "stroke": "black",
        "strokeWidth": "2",
        "width": 1024,
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
